src/main/kotlin/dev/znci/rocket/Rocket.kt

```diff
@@ -17,15 +17,13 @@ package dev.znci.rocket
 
 import dev.znci.rocket.commands.RocketCommand
 import dev.znci.rocket.i18n.LocaleManager
+import dev.znci.rocket.scripting.AddonManager
 import dev.znci.rocket.scripting.ScriptManager
 import dev.znci.rocket.scripting.events.EventListener
-import dev.znci.rocket.scripting.GlobalInitializer
-import dev.znci.rocket.scripting.annotations.Global
-import org.bukkit.plugin.java.JavaPlugin
-import org.reflections.Reflections
+import dev.znci.rocket.scripting.api.RocketAddon
 import java.io.File
 
-open class Rocket : JavaPlugin() {
+open class Rocket : RocketAddon() {
```

Comment:
I'm not sure how much I like us not directly extending JavaPlugin. Not the class hierarchy reasons (Paper's code guidelines always want you base class to be JavaPlugin, but I do not care at all about Paper's guidelines), but for code style reasons. **TL;DR: _Rocket itself is technically not an addon_.** 

I'm generally okay with this, but maybe we can rename it to RocketPlugin?
Or, rename the _current_ RocketAddon to RocketPlugin, and create a _new_ RocketAddon that only extends RocketPlugin. This will also make it easier to "tag" addons as addons in the future. It may also give us the ability to disable misbehaving addons/incompatible addons.

I defer to you, @zNotChill, on what to do here, but I think it is a point of discussion :slightly_smiling_face:

Reply:
It made it easier for us since Rocket **acts** as an addon, because the RocketAddon already has auto registering globals, so it made sense to me to convert the main Rocket.kt to use a RocketAddon so this is taken care of for us.

Addons would just pretty much follow the exact same styling and classes as the main Rocket plugin (see my example)

You are right about rocket not technically being an addon, but it makes sense to treat it as one, since it makes our job easier when making changes to rocket and other RocketAddon modifications